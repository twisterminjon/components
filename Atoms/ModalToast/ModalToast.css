.modaltoast-wrap {
  /* This required to handle the height animation, w/o the padding on the content will show  */
  position: fixed;
  width: 375px;
  height: 0;
  overflow-y: hidden;

  top: 0;
  left: 50%;
  transform: translateX(-50%);

  transition: height 200ms ease-out;
  z-index: var(--z-modal);
}

.modaltoast-dimmer {
  position: fixed;
  background: rgba(0, 0, 0, 0.75);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  z-index: calc(var(--z-modal) - 1);
}

.modaltoast {
  position: relative;
  color: white;

  width: 375px;
  padding: 16px 16px 30px 16px;

  background: var(--toastmodal_bg);

  border-bottom-left-radius: 38px;
  border-bottom-right-radius: 38px;
}

.modaltoast--show {
  height: 100%;
  transition: height 200ms ease-in;
  overflow: visible;
}

.modaltoastheader h3 {
  color: var(--white);
  font-size: 20px;
  font-weight: 300;
  margin-bottom: 15px !important;
}

.modaltoasttext {
  width: 100%;
  text-align: center;
  white-space: pre-wrap;
  color: var(--brandcolor);
  font-size: 14px;
  font-weight: 300;
  margin-bottom: 15px !important;
}

.modaltoastbutton {
  margin-bottom: 5px !important;
}
