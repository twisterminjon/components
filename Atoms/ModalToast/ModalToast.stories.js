import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs, boolean, text } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import ModalToast from './ModalToast'

const stories = storiesOf('Provider/v1/Atoms', module)
stories.addDecorator(withKnobs)

stories.add('ModalToast', () => {
  const showLab = 'show'
  const showDef = true
  const showVal = boolean(showLab, showDef)

  return <ModalToast show={showVal} />
})

stories.add('ModalToast with Header and text', () => {
  const showLab = 'show'
  const showDef = true
  const showVal = boolean(showLab, showDef)
  const titleVal = text('title', 'This is the Title in ModalToast.Header')
  const textVal = text('text', 'This is the text in ModalToast.Text')
  const textVal2 = text('text2', 'This is another child')
  const buttonLabel = text('buttonLabel', 'Button with ModalToast.Button')

  return (
    <ModalToast show={showVal}>
      <ModalToast.Header title={titleVal} />
      <ModalToast.Text>
        <span style={{ color: 'var(--white)', fontWeight: 'bold' }}>{textVal}</span>
        {textVal2}
      </ModalToast.Text>
      <ModalToast.Button buttonLabel={buttonLabel} fluid onClick={action('Click on Button')} />
      <ModalToast.Button isGhostButton={true} onClick={action('Click on Button')} />
    </ModalToast>
  )
})
