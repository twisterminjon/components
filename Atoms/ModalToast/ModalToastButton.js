import React from 'react'
import PropTypes from 'prop-types'
import Button from '../../Atoms/Button/Button'
import ButtonGhost from '../../Atoms/ButtonGhost/ButtonGhost'

import './ModalToast.css'

export default function ModalToastButton({
  isGhostButton = false,
  buttonLabel = 'Close',
  onClick = () => {},
  className = '',
  ...rest
}) {
  const testId = rest['data-testid'] ? rest['data-testid'] : `modaltoastbutton-${buttonLabel}`

  if (isGhostButton)
    return (
      <ButtonGhost
        className={`modaltoastbutton ${className}`.trim()}
        onClick={onClick}
        data-testid={testId}
        autoFocus={true}
        {...rest}>
        {buttonLabel}
      </ButtonGhost>
    )

  return (
    <Button
      className={`modaltoastbutton ${className}`.trim()}
      onClick={onClick}
      data-testid={testId}
      autoFocus={true}
      {...rest}>
      {buttonLabel}
    </Button>
  )
}

ModalToastButton.propTypes = {
  /** If set then it will render a ghost button, if not a regular button */
  isGhostButton: PropTypes.bool,
  /** Label for the Button */
  buttonLabel: PropTypes.string,
  /** Function to be called on click */
  onClick: PropTypes.func,
}
