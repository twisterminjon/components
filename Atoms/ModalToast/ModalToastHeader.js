import React from 'react'
import PropTypes from 'prop-types'

import './ModalToast.css'

export default function ModalToastHeader({ title, className = '', style = {}, ...rest }) {
  const testId = rest['data-testid'] ? rest['data-testid'] : 'modaltoastheader'

  return (
    <div className={`modaltoastheader ${className}`.trim()} data-testid={testId}>
      <h3 style={style} {...rest}>
        {title}
      </h3>
    </div>
  )
}

ModalToastHeader.propTypes = {
  /** Title header for the ModalToast Header*/
  title: PropTypes.string.isRequired,
}
