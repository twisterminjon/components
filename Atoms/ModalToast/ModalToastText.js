import React from 'react'
import PropTypes from 'prop-types'

import './ModalToast.css'

export default function ModalToastText({ children, className = '', style = {}, ...rest }) {
  const testId = rest['data-testid'] ? rest['data-testid'] : 'modaltoasttext'
  return (
    <p className={`modaltoasttext ${className}`.trim()} data-testid={testId} style={style} {...rest}>
      {children}
    </p>
  )
}

ModalToastText.propTypes = {
  /** Nodes to be rendered in the ModalToastText */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
}
