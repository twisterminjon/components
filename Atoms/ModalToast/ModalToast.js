import React from 'react'
import PropTypes from 'prop-types'
import ModalToastHeader from './ModalToastHeader'
import ModalToastText from './ModalToastText'
import ModalToastButton from './ModalToastButton'

import './ModalToast.css'

export default function ModalToast(props) {
  const { show, children } = props
  const visibilityClass = show ? 'modaltoast--show' : ''

  const testId = props['data-testid'] ? props['data-testid'] : 'modaltoast'

  return (
    <React.Fragment>
      {show && <div className="modaltoast-dimmer" />}
      <div className={`modaltoast-wrap ${visibilityClass}`.trim()} data-testid={testId}>
        <div className={`modaltoast`}>{children}</div>
      </div>
    </React.Fragment>
  )
}

ModalToast.propTypes = {
  /** The model can be shown or hidden */
  show: PropTypes.bool.isRequired,
}

ModalToast.Header = ModalToastHeader
ModalToast.Text = ModalToastText
ModalToast.Button = ModalToastButton
