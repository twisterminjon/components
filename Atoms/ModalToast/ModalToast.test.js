import React from 'react'
import ModalToast from './ModalToast'

describe('ModalToast', () => {
  const mockFunction = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<ModalToast show={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('uses a custom testid', () => {
    const wrapper = window.shallow(<ModalToast show={true} data-testid="custom" />)
    expect(wrapper.find('[data-testid="custom"]')).toBeDefined()
  })

  it('has the show class when shown', () => {
    const wrapper = window.shallow(<ModalToast show={true} />)
    const showPart = wrapper.find('[data-testid="modaltoast"]')
    expect(showPart.hasClass('modaltoast--show')).toBeTruthy()
  })

  it('show=false removes the show class', () => {
    const wrapper = window.shallow(<ModalToast show={false} />)
    const showPart = wrapper.find('[data-testid="modaltoast"]')
    expect(showPart.hasClass('modaltoast--show')).toBeFalsy()
  })

  it('matches the snapshot with a Header, Text, Button', () => {
    const wrapper = window.shallow(
      <ModalToast show={true}>
        <ModalToast.Header title={`Title`} />
        <ModalToast.Text>some text within a paragraph</ModalToast.Text>
        <ModalToast.Button buttonLabel={'close'} fluid onClick={mockFunction} />
        <ModalToast.Button fluid isGhostButton={true} onClick={mockFunction} />
      </ModalToast>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
