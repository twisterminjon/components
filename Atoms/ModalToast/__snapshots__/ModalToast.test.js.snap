// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalToast matches the snapshot 1`] = `
<Fragment>
  <div
    className="modaltoast-dimmer"
  />
  <div
    className="modaltoast-wrap modaltoast--show"
    data-testid="modaltoast"
  >
    <div
      className="modaltoast"
    />
  </div>
</Fragment>
`;

exports[`ModalToast matches the snapshot with a Header, Text, Button 1`] = `
<Fragment>
  <div
    className="modaltoast-dimmer"
  />
  <div
    className="modaltoast-wrap modaltoast--show"
    data-testid="modaltoast"
  >
    <div
      className="modaltoast"
    >
      <ModalToastHeader
        title="Title"
      />
      <ModalToastText>
        some text within a paragraph
      </ModalToastText>
      <ModalToastButton
        buttonLabel="close"
        fluid={true}
        onClick={[MockFunction]}
      />
      <ModalToastButton
        fluid={true}
        isGhostButton={true}
        onClick={[MockFunction]}
      />
    </div>
  </div>
</Fragment>
`;
